packages/react-core/__tests__/operations/groupByDate.test.js

```diff
@@ -0,0 +1,149 @@
+import { AggregationTypes } from '../../src/operations/aggregation/AggregationTypes';
+import { GroupDateTypes } from '../../src/operations/GroupDateTypes';
+import { groupValuesByDateColumn } from '../../src/operations/groupByDate';
+
+const REVENUE = 50;
+
+const DATES_VALUES = [
+  Date.UTC(1970, 0, 1, 0, 0),
+  Date.UTC(1970, 0, 1, 0, 30),
+  Date.UTC(1970, 1, 1, 0, 0),
+  Date.UTC(1970, 1, 1, 0, 30),
+  Date.UTC(1971, 0, 1, 1, 0)
+];
+
+const DATE_COLUMN = 'test_date';
+const OPERATION_COLUMN = 'test_revenue';
+
+const FEATURES = DATES_VALUES.map((value) => ({
+  [DATE_COLUMN]: value,
+  [OPERATION_COLUMN]: REVENUE
+}));
+
+describe('groupValuesByDateColumn', () => {
+  test('should return null due to empty data array', () => {
+    expect(groupValuesByDateColumn([])).toEqual(null);
+  });
+
+  describe('valid features', () => {
+    test('should return null due to invalid grouping operation', () => {
+      expect(
+        groupValuesByDateColumn(
+          FEATURES,
+          OPERATION_COLUMN,
+          DATE_COLUMN,
+          '__fake_group_type__',
+          AggregationTypes.COUNT
+        )
+      ).toEqual(null);
+    });
+
+    test('should return an empty array due to invalid operation', () => {
+      expect(
+        groupValuesByDateColumn(
+          FEATURES,
+          OPERATION_COLUMN,
+          DATE_COLUMN,
+          GroupDateTypes.DAYS,
+          '__fake_operation__'
+        )
+      ).toEqual([]);
+    });
+
+    describe('grouping operation tests', () => {
+      const ARGS = [FEATURES, OPERATION_COLUMN, DATE_COLUMN, GroupDateTypes.YEARS];
+
+      test(GroupDateTypes.YEARS, () => {
+        const RESULT = [
+          { name: Date.UTC(1970, 0, 1), value: 4 },
+          { name: Date.UTC(1971, 0, 1), value: 1 }
+        ];
+
+        executeGroupByDateFnTests(ARGS, RESULT);
+      });
+
+      test(GroupDateTypes.MONTHS, () => {
+        ARGS[3] = GroupDateTypes.MONTHS;
```

Comment:
It would be probably easier to read if your ARGS is a dictionary, not array, and you do 

```
const params = { features, operation, date, grouping }
// ...
params.grouping = XXX
// ...
executeGroupByDateFnTests(params, RESULT);
```

Reply:
Much better! I changed it :)